aws_lambda_builders/workflows/rust_cargo/actions.py

```diff
@@ -0,0 +1,216 @@
+"""
+Rust Cargo build actions
+"""
+
+import os
+import subprocess
+import shutil
+import json
+
+from aws_lambda_builders.workflow import BuildMode
+from aws_lambda_builders.actions import ActionFailedError, BaseAction, Purpose
+
+
+class BuilderError(Exception):
+    MESSAGE = "Builder Failed: {message}"
+
+    def __init__(self, **kwargs):
+        Exception.__init__(self, self.MESSAGE.format(**kwargs))
+
+
+class OSUtils(object):
```

Comment:
A comment from a first time contributor. These utils while helpful seem to be duplicated for just about builder that does something with popen. Perhaps it would be useful to extract a common util so that new builders didn't need to continue the copy process.

Reply:
Thanks.

> Over time, it just became 'easier' but we should extract these out.

The irony is it made it harder for me. I had to copy util code and in doing so had to copy the tests for that util code to uphold this projects codecov requirements. 

> If you want to do it, I would open up a new PR 

 I'd plotted out the time to set aside to get these changes worked through and estimate the time to get this integrated into the separate sam cli repo before starting this journey.

Given the review turnaround time, I don't anticipate being able to invest additional time in cleaning up this project's existing tech debt. It would be a good investment on behalf of the project maintainers though. I just wanted to capture in a note some of the experience from a first time contributor incase that was useful to project maintainers.